src/loaders/external_png/tvgPngLoader.cpp

```diff
@@ -110,16 +130,20 @@ bool PngLoader::close()
     return true;
 }
 
-unique_ptr<Surface> PngLoader::bitmap()
+unique_ptr<Surface> PngLoader::bitmap(int32_t colorSpace)
 {
     if (!content) return nullptr;
+    if (colorSpace != -1 && static_cast<uint32_t>(colorSpace) != this->colorSpace) {
```

Comment:
This converting compatibility looks quite fragile. At least you can add a description here(other loaders as well) what it does and how this must be taken over the possible coming other colorspace types.